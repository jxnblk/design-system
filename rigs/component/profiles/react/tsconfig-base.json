{
  "$schema": "http://json.schemastore.org/tsconfig",

  "compilerOptions": {
    "outDir": "../../../../../lib-esm",
    "rootDir": "../../../../../src",
    "rootDirs": ["../../../../../src", "../../../../../temp/sass-ts"],

    "forceConsistentCasingInFileNames": true,
    "jsx": "react",
    "declaration": true,
    "sourceMap": true,
    "declarationMap": true,
    "inlineSources": true,
    "experimentalDecorators": true,

    "esModuleInterop": true,
    "noEmitOnError": false,
    "allowUnreachableCode": false,

    "module": "es2015",
    "moduleResolution": "node",
    "target": "es5",
    "lib": ["es5", "scripthost", "es2015.collection", "es2015.promise", "es2015.iterable", "dom", "es2016"],

    // deviations from @rushstack/heft-web-rig/profiles/library/tsconfig-base.json
    "strict": false,

    "types": [
      "../../rigs/component/node_modules/@types/heft-jest",
      "../../rigs/component/node_modules/@types/node"
    ],

    "allowJs": true
  },
  "include": ["../../../../../src/**/*"],
  "exclude": ["node_modules", "lib-esm"]
}
