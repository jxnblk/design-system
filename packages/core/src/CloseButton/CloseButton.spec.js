import React from 'react'
import { fireEvent } from '@testing-library/react'

import { CloseButton } from '..'

describe('CloseButton', () => {
  test('executes onClick prop on click', () => {
    const handleClick = jest.fn()
    const { container } = renderWithTheme(<CloseButton onClick={handleClick} />)
    fireEvent.click(container.firstChild)
    expect(handleClick).toBeCalled()
  })

  test('renders without props', () => {
    const json = rendererCreateWithTheme(<CloseButton />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
