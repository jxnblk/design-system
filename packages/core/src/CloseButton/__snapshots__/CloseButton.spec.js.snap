// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders without props 1`] = `
.c1 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  width: 24px;
}

.c2 {
  outline: none;
}

.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  padding: 0;
  background-color: transparent;
  color: inherit;
}

.c0:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c0:hover {
  background-color: transparent;
  color: inherit;
}

.c0 > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<button
  className="c0"
  size={24}
  title="close"
>
  <div>
    <svg
      aria-hidden={true}
      className="c1 c2"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M19 6.4L17.6 5 12 10.6 6.4 5 5 6.4l5.6 5.6L5 17.6 6.4 19l5.6-5.6 5.6 5.6 1.4-1.4-5.6-5.6L19 6.4z"
      />
    </svg>
  </div>
</button>
`;
